### ISSUE TITLE
My install was much more verbose than yours

### ISSUE THREAD

**Issue by JoshCheek** [trptcolin/object-vector-is-not-a-value#1]
- Works for me, but output of `sbt clean run` is different:
  
  <details><summary>
  
  <pre>$ sbt clean run</pre></summary>
  
  
  
  ```
  Getting org.scala-sbt sbt 0.13.5 ...
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt/0.13.5/jars/sbt.jar ...
    [SUCCESSFUL ] org.scala-sbt#sbt;0.13.5!sbt.jar (1649ms)
  downloading https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.10.4/scala-library-2.10.4.jar ...
    [SUCCESSFUL ] org.scala-lang#scala-library;2.10.4!scala-library.jar (3302ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/main/0.13.5/jars/main.jar ...
    [SUCCESSFUL ] org.scala-sbt#main;0.13.5!main.jar (3100ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compiler-interface/0.13.5/jars/compiler-interface-src.jar ...
    [SUCCESSFUL ] org.scala-sbt#compiler-interface;0.13.5!compiler-interface-src.jar (1142ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compiler-interface/0.13.5/jars/compiler-interface-bin.jar ...
    [SUCCESSFUL ] org.scala-sbt#compiler-interface;0.13.5!compiler-interface-bin.jar (1334ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/precompiled-2_8_2/0.13.5/jars/compiler-interface-bin.jar ...
    [SUCCESSFUL ] org.scala-sbt#precompiled-2_8_2;0.13.5!compiler-interface-bin.jar (2333ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/precompiled-2_9_2/0.13.5/jars/compiler-interface-bin.jar ...
    [SUCCESSFUL ] org.scala-sbt#precompiled-2_9_2;0.13.5!compiler-interface-bin.jar (2810ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/precompiled-2_9_3/0.13.5/jars/compiler-interface-bin.jar ...
    [SUCCESSFUL ] org.scala-sbt#precompiled-2_9_3;0.13.5!compiler-interface-bin.jar (1466ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/actions/0.13.5/jars/actions.jar ...
    [SUCCESSFUL ] org.scala-sbt#actions;0.13.5!actions.jar (1886ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/main-settings/0.13.5/jars/main-settings.jar ...
    [SUCCESSFUL ] org.scala-sbt#main-settings;0.13.5!main-settings.jar (2288ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/interface/0.13.5/jars/interface.jar ...
    [SUCCESSFUL ] org.scala-sbt#interface;0.13.5!interface.jar (1191ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/io/0.13.5/jars/io.jar ...
    [SUCCESSFUL ] org.scala-sbt#io;0.13.5!io.jar (1981ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/ivy/0.13.5/jars/ivy.jar ...
    [SUCCESSFUL ] org.scala-sbt#ivy;0.13.5!ivy.jar (2400ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/launcher-interface/0.13.5/jars/launcher-interface.jar ...
    [SUCCESSFUL ] org.scala-sbt#launcher-interface;0.13.5!launcher-interface.jar (1166ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/logging/0.13.5/jars/logging.jar ...
    [SUCCESSFUL ] org.scala-sbt#logging;0.13.5!logging.jar (1254ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/logic/0.13.5/jars/logic.jar ...
    [SUCCESSFUL ] org.scala-sbt#logic;0.13.5!logic.jar (1136ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/process/0.13.5/jars/process.jar ...
    [SUCCESSFUL ] org.scala-sbt#process;0.13.5!process.jar (1137ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/run/0.13.5/jars/run.jar ...
    [SUCCESSFUL ] org.scala-sbt#run;0.13.5!run.jar (1129ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/command/0.13.5/jars/command.jar ...
    [SUCCESSFUL ] org.scala-sbt#command;0.13.5!command.jar (1570ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/classpath/0.13.5/jars/classpath.jar ...
    [SUCCESSFUL ] org.scala-sbt#classpath;0.13.5!classpath.jar (1231ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/completion/0.13.5/jars/completion.jar ...
    [SUCCESSFUL ] org.scala-sbt#completion;0.13.5!completion.jar (2037ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/api/0.13.5/jars/api.jar ...
    [SUCCESSFUL ] org.scala-sbt#api;0.13.5!api.jar (1377ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compiler-integration/0.13.5/jars/compiler-integration.jar ...
    [SUCCESSFUL ] org.scala-sbt#compiler-integration;0.13.5!compiler-integration.jar (1183ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compiler-ivy-integration/0.13.5/jars/compiler-ivy-integration.jar ...
    [SUCCESSFUL ] org.scala-sbt#compiler-ivy-integration;0.13.5!compiler-ivy-integration.jar (1112ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/relation/0.13.5/jars/relation.jar ...
    [SUCCESSFUL ] org.scala-sbt#relation;0.13.5!relation.jar (1114ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/task-system/0.13.5/jars/task-system.jar ...
    [SUCCESSFUL ] org.scala-sbt#task-system;0.13.5!task-system.jar (1219ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/tasks/0.13.5/jars/tasks.jar ...
    [SUCCESSFUL ] org.scala-sbt#tasks;0.13.5!tasks.jar (1224ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/tracking/0.13.5/jars/tracking.jar ...
    [SUCCESSFUL ] org.scala-sbt#tracking;0.13.5!tracking.jar (1235ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/testing/0.13.5/jars/testing.jar ...
    [SUCCESSFUL ] org.scala-sbt#testing;0.13.5!testing.jar (1220ms)
  downloading https://repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.10.4/scala-compiler-2.10.4.jar ...
    [SUCCESSFUL ] org.scala-lang#scala-compiler;2.10.4!scala-compiler.jar (5359ms)
  downloading https://repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.10.4/scala-reflect-2.10.4.jar ...
    [SUCCESSFUL ] org.scala-lang#scala-reflect;2.10.4!scala-reflect.jar (1338ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/control/0.13.5/jars/control.jar ...
    [SUCCESSFUL ] org.scala-sbt#control;0.13.5!control.jar (1224ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/collections/0.13.5/jars/collections.jar ...
    [SUCCESSFUL ] org.scala-sbt#collections;0.13.5!collections.jar (3342ms)
  downloading https://repo1.maven.org/maven2/jline/jline/2.11/jline-2.11.jar ...
    [SUCCESSFUL ] jline#jline;2.11!jline.jar (134ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/incremental-compiler/0.13.5/jars/incremental-compiler.jar ...
    [SUCCESSFUL ] org.scala-sbt#incremental-compiler;0.13.5!incremental-compiler.jar (2596ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/compile/0.13.5/jars/compile.jar ...
    [SUCCESSFUL ] org.scala-sbt#compile;0.13.5!compile.jar (1193ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/persist/0.13.5/jars/persist.jar ...
    [SUCCESSFUL ] org.scala-sbt#persist;0.13.5!persist.jar (1147ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/classfile/0.13.5/jars/classfile.jar ...
    [SUCCESSFUL ] org.scala-sbt#classfile;0.13.5!classfile.jar (1123ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-tools.sbinary/sbinary_2.10/0.4.2/jars/sbinary_2.10.jar ...
    [SUCCESSFUL ] org.scala-tools.sbinary#sbinary_2.10;0.4.2!sbinary_2.10.jar (1118ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/cross/0.13.5/jars/cross.jar ...
    [SUCCESSFUL ] org.scala-sbt#cross;0.13.5!cross.jar (1091ms)
  downloading https://repo1.maven.org/maven2/org/apache/ivy/ivy/2.3.0/ivy-2.3.0.jar ...
    [SUCCESSFUL ] org.apache.ivy#ivy;2.3.0!ivy.jar (522ms)
  downloading https://repo1.maven.org/maven2/com/jcraft/jsch/0.1.46/jsch-0.1.46.jar ...
    [SUCCESSFUL ] com.jcraft#jsch;0.1.46!jsch.jar (128ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/cache/0.13.5/jars/cache.jar ...
    [SUCCESSFUL ] org.scala-sbt#cache;0.13.5!cache.jar (1217ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/test-agent/0.13.5/jars/test-agent.jar ...
    [SUCCESSFUL ] org.scala-sbt#test-agent;0.13.5!test-agent.jar (1105ms)
  downloading https://repo1.maven.org/maven2/org/scala-sbt/test-interface/1.0/test-interface-1.0.jar ...
    [SUCCESSFUL ] org.scala-sbt#test-interface;1.0!test-interface.jar (86ms)
  downloading https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/apply-macro/0.13.5/jars/apply-macro.jar ...
    [SUCCESSFUL ] org.scala-sbt#apply-macro;0.13.5!apply-macro.jar (1191ms)
  :: retrieving :: org.scala-sbt#boot-app
    confs: [default]
    44 artifacts copied, 0 already retrieved (13482kB/138ms)
  Getting Scala 2.10.4 (for sbt)...
  downloading https://repo1.maven.org/maven2/org/scala-lang/jline/2.10.4/jline-2.10.4.jar ...
    [SUCCESSFUL ] org.scala-lang#jline;2.10.4!jline.jar (158ms)
  downloading https://repo1.maven.org/maven2/org/fusesource/jansi/jansi/1.4/jansi-1.4.jar ...
    [SUCCESSFUL ] org.fusesource.jansi#jansi;1.4!jansi.jar (123ms)
  :: retrieving :: org.scala-sbt#boot-scala
    confs: [default]
    5 artifacts copied, 0 already retrieved (24459kB/116ms)
  [info] Loading project definition from /Users/josh/deleteme/object-vector-is-not-a-value/project
  [info] Updating {file:/Users/josh/deleteme/object-vector-is-not-a-value/project/}object-vector-is-not-a-value-build...
  [info] Resolving org.fusesource.jansi#jansi;1.4 ...
  [info] Done updating.
  [info] Set current project to hello (in build file:/Users/josh/deleteme/object-vector-is-not-a-value/)
  [success] Total time: 0 s, completed Oct 11, 2016 6:07:19 PM
  [info] Updating {file:/Users/josh/deleteme/object-vector-is-not-a-value/}object-vector-is-not-a-value...
  [info] Resolving org.sonatype.oss#oss-parent;7 ...
  [info] downloading http://repo1.maven.org/maven2/org/scala-lang/scala-library/2.11.8/scala-library-2.11.8.jar ...
  [info]    [SUCCESSFUL ] org.scala-lang#scala-library;2.11.8!scala-library.jar (1785ms)
  [info] downloading http://repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.11.8/scala-compiler-2.11.8.jar ...
  [info]    [SUCCESSFUL ] org.scala-lang#scala-compiler;2.11.8!scala-compiler.jar (4487ms)
  [info] downloading http://repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.11.8/scala-reflect-2.11.8.jar ...
  [info]    [SUCCESSFUL ] org.scala-lang#scala-reflect;2.11.8!scala-reflect.jar (1460ms)
  [info] downloading http://repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.11/1.0.4/scala-xml_2.11-1.0.4.jar ...
  [info]    [SUCCESSFUL ] org.scala-lang.modules#scala-xml_2.11;1.0.4!scala-xml_2.11.jar(bundle) (242ms)
  [info] downloading http://repo1.maven.org/maven2/org/scala-lang/modules/scala-parser-combinators_2.11/1.0.4/scala-parser-combinators_2.11-1.0.4.jar ...
  [info]    [SUCCESSFUL ] org.scala-lang.modules#scala-parser-combinators_2.11;1.0.4!scala-parser-combinators_2.11.jar(bundle) (198ms)
  [info] downloading http://repo1.maven.org/maven2/jline/jline/2.12.1/jline-2.12.1.jar ...
  [info]    [SUCCESSFUL ] jline#jline;2.12.1!jline.jar (159ms)
  [info] Done updating.
  [info] Compiling 1 Scala source to /Users/josh/deleteme/object-vector-is-not-a-value/target/scala-2.11/classes...
  [info] 'compiler-interface' not yet compiled for Scala 2.11.8. Compiling...
  [info]   Compilation completed in 10.617 s
  [info] Running Hi
  ohai!
  kthxbai!
  [success] Total time: 21 s, completed Oct 11, 2016 6:07:40 PM
  ```
  
  </details>
  
  Running a second time with the `-v` (verbose) and `-g` (debug) flags on:
  
  <details><summary>
  
  <pre>$ sbt clean run -v -g</pre></summary>
  
  
  
  ```
  [process_args] java_version = '1.8.0_71'
  # Executing command line:
  java
  -Xms1024m
  -Xmx1024m
  -XX:ReservedCodeCacheSize=128m
  -XX:MaxMetaspaceSize=256m
  -jar
  /usr/local/Cellar/sbt/0.13.12/libexec/sbt-launch.jar
  clean
  run
  -g
  
  [info] Loading project definition from /Users/josh/deleteme/object-vector-is-not-a-value/project
  [info] Set current project to hello (in build file:/Users/josh/deleteme/object-vector-is-not-a-value/)
  [success] Total time: 0 s, completed Oct 11, 2016 6:35:58 PM
  [info] Updating {file:/Users/josh/deleteme/object-vector-is-not-a-value/}object-vector-is-not-a-value...
  [info] Resolving jline#jline;2.12.1 ...
  [info] Done updating.
  [info] Compiling 1 Scala source to /Users/josh/deleteme/object-vector-is-not-a-value/target/scala-2.11/classes...
  [info] Running Hi
  ohai!
  kthxbai!
  [success] Total time: 4 s, completed Oct 11, 2016 6:36:02 PM
  [warn] The `-` command is deprecated in favor of `onFailure` and will be removed in 0.14.0
  ```
  
  </details>
- I can't figure out how to ask what my sbt version is -.-
- When I try to explicitly set it to the version listed in the `build.sbt`, it blows up at me
  <details><summary>
  
  <pre>$ sbt -sbt-version 2.11.8 clean run</pre></summary>
  
  
  
  ```
  Getting org.scala-sbt sbt 2.11.8 ...
  
  :: problems summary ::
  :::: WARNINGS
        module not found: org.scala-sbt#sbt;2.11.8
  
    ==== local: tried
  
      /Users/josh/.ivy2/local/org.scala-sbt/sbt/2.11.8/ivys/ivy.xml
  
      -- artifact org.scala-sbt#sbt;2.11.8!sbt.jar:
  
      /Users/josh/.ivy2/local/org.scala-sbt/sbt/2.11.8/jars/sbt.jar
  
    ==== Maven Central: tried
  
      https://repo1.maven.org/maven2/org/scala-sbt/sbt/2.11.8/sbt-2.11.8.pom
  
      -- artifact org.scala-sbt#sbt;2.11.8!sbt.jar:
  
      https://repo1.maven.org/maven2/org/scala-sbt/sbt/2.11.8/sbt-2.11.8.jar
  
    ==== typesafe-ivy-releases: tried
  
      https://repo.typesafe.com/typesafe/ivy-releases/org.scala-sbt/sbt/2.11.8/ivys/ivy.xml
  
    ==== sbt-ivy-snapshots: tried
  
      https://repo.scala-sbt.org/scalasbt/ivy-snapshots/org.scala-sbt/sbt/2.11.8/ivys/ivy.xml
  
        ::::::::::::::::::::::::::::::::::::::::::::::
  
        ::          UNRESOLVED DEPENDENCIES         ::
  
        ::::::::::::::::::::::::::::::::::::::::::::::
  
        :: org.scala-sbt#sbt;2.11.8: not found
  
        ::::::::::::::::::::::::::::::::::::::::::::::
  
  
  
  :: USE VERBOSE OR DEBUG MESSAGE LEVEL FOR MORE DETAILS
  unresolved dependency: org.scala-sbt#sbt;2.11.8: not found
  Error during sbt execution: Error retrieving required libraries
    (see /Users/josh/.sbt/boot/update.log for complete log)
  Error: Could not retrieve sbt 2.11.8
  ```
  
  </details> 
- My Java version is `1.8.0_71`
  
  <details><summary>
  
  <pre>$ java -version</pre></summary>
  
  
  
  ```
  java version "1.8.0_71"
  Java(TM) SE Runtime Environment (build 1.8.0_71-b15)
  Java HotSpot(TM) 64-Bit Server VM (build 25.71-b15, mixed mode)
  ```
  
  </details>
- My Scala version is 2.11.8
  
  <details><summary>
  
  <pre>$ scala -version</pre></summary>
  
  
  
  ```
  Scala code runner version 2.11.8 -- Copyright 2002-2016, LAMP/EPFL
  ```
  
  </details>
  
  Weirdly, however, when I look in `~/.sbt`, it lists `2.10.4`
  
  <details><summary>
  
  <pre>$ ls -l ~/.sbt/boot/</pre></summary>
  
  
  
  ```
  total 40
  -rw-r--r--  1 josh  staff      0 Oct 11 18:35 sbt.boot.lock
  drwxr-xr-x  4 josh  staff    136 Oct 11 18:07 scala-2.10.4
  -rw-r--r--  1 josh  staff  18645 Oct 11 18:30 update.log
  ```
  
  </details>
- I don't have any env vars set for Java/Scala/Sbt (no output)
  
  ```
  $ env | ag -i 'java|scala|sbt'
  ```
- My global config file has `-sbt-create` set
  <details><summary>
  
  <pre>$ sed -e '/^$/d' -e '/^#/d' < /usr/local/etc/sbtopts</pre></summary>
  
  
  `
  -sbt-create
  `
  </details>

Ideas:
- Try using the `-v` (verbose) and `-d` anything weird showing up?
- Make sure your env vars are similarly empty
- Make sure your global config file matches mine
- Perhaps you already had a dep installed and thus it installed a newer one on mine?
- Perhaps you've modified a built in / cached value (eg have you installed custom versions of packages above, eg sometimes I do stuff like `vim "$(gem which bundler/setup)"` and forget to remove my edits, you might've done something like that, eg have you installed a custom jline or anything?
- Ignoring any of the above, the error sounds to me like it's interpreting that line differently. I know nothing about Scala syntax, but it seems strange to create a vector that way (eg it looks like vector doesn't have its own syntax and rather is taking advantage of a macro named `Vector` or, in Ruby, would be taking advantage of Ruby's parsing it into rest arguments) perhaps there was a change to the parser at some point, causing this line to be treated differently?
- You said that if you add a package, it fixes the issue, which package? Does it have a dependency on Vector which changes how it resolves the symbol? (eg changing the class path or importing some core package into the environment that is not otherwise there by default for you, for some reason).

**Comment by trptcolin** [comment]
Thanks for taking a look, and for showing a few settings. I've scoped the problem down further now, to a `scalac` invocation without involving SBT at all (it's at the bottom of the README now, just `scalac src/main/scala/hw.scala`). I can also see it working in a local VM.

The `Vector(1, 2, 3)` invocation _should_ be fine (and is on many others' machines, clearly. It works  for me in the `scala` REPL (basically it maps to an invocation of the `apply` method on the `Vector` companion object).

Added more notes to the README to clarify the "add a package" thing - I mean "add a package declaration" (see the commented-out package declaration in `src/main/scala/hw.scala`).

Maybe it's just undefined behavior to not put a package on your code - I wasn't able to find anything in the scala spec about the default package. This whole thing is extracted from a tutorial project by somebody who knows scala really well though, which makes me assume there's something wrong in my setup. 😿
